File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/UserViewModel.java
package mx.edu.upqroo.kristenandroid.viewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import mx.edu.upqroo.kristenandroid.data.database.entities.UserInformation;
import mx.edu.upqroo.kristenandroid.data.repositories.UserInformationRepository;

public class UserViewModel extends AndroidViewModel {
    private UserInformationRepository mUserInfoRepo;
    private LiveData<UserInformation> mUserInfo;

    public UserViewModel(@NonNull Application application) {
        super(application);
        mUserInfoRepo = UserInformationRepository.getInstance(application);
    }

    public LiveData<UserInformation> getUser(@NonNull String id) {
        return mUserInfoRepo.getUserInformation(id);
    }

    public void insert(UserInformation userInformation) {
        mUserInfoRepo.insert(userInformation);
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/UserViewModel.kt
package mx.edu.upqroo.kristenandroid.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import mx.edu.upqroo.kristenandroid.data.database.entities.UserInformation
import mx.edu.upqroo.kristenandroid.data.repositories.UserInformationRepository

class UserViewModel(application: Application) : AndroidViewModel(application) {
    private val mUserInfoRepo: UserInformationRepository =
            UserInformationRepository.getInstance(application)

    fun getUser(id: String): LiveData<UserInformation> {
        return mUserInfoRepo.getUserInformation(id)
    }

    fun insert(userInformation: UserInformation) {
        mUserInfoRepo.insert(userInformation)
    }
}
